evaluator/images/java/entry.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/local/bin/python3
+import dataclasses
+import os
+import subprocess
+import json
+import re
+import glob
+from collections import defaultdict
+from pathlib import Path
+from typing import Dict, List, Optional
+import xml.etree.ElementTree as ET
+
+import bleach
+
+
+@dataclasses.dataclass
+class BuildResult:
+    success: bool
+    output: str
+    comments: Optional[Dict[str, List[str]]] = dataclasses.field(default_factory=dict)
+    tests: Optional[List[dict]] = dataclasses.field(default_factory=list)
+
+    @staticmethod
+    def fail(error: str) -> "BuildResult":
+        return BuildResult(
+            success=False,
+            output=error,
+        )
+
+
+def parse_tests_report(path):
+    try:
+        for path in glob.glob(os.path.join(path, "*.xml")):
+            tree = ET.parse(path)
+            os.unlink(path)
+
+            tests = []
+            for testcase in tree.findall('.//testcase'):
+                success = True
+                message = ""
+                failure = testcase.find('./failure')
+                
+                if failure is not None:
+                    success = False
+                    message = failure.get('message', '') + "\n" + (failure.text or '').strip()
+                    
+                tests.append({
+                    'name': testcase.get('name'),
+                    'success': success,
+                    'message': message.strip(),
+                })
+
+        return tests
+    except FileNotFoundError:
+        return []
+
+
+def get_executable_class_names(directory: Path) -> List[str]:
+    def find_java_files(directory):
+        for root, dirs, files in os.walk(directory):
+            for file in files:
+                if file.endswith(".java"):
+                    yield os.path.join(root, file)
+
+    def is_executable_java_file(file_path):
+        with open(file_path, 'r', encoding='utf-8', errors='ignore') as file:
+            content = file.read()
+            # Vzor pro nalezení metody main v Java souboru
+            pattern = re.compile(r"public\s+static\s+void\s+main\s*\(\s*String\s*\[\]\s+args\s*\)")
+            return bool(re.search(pattern, content))
+
+    def get_qualified_class_name(file_path, base_directory):
+        # Odstranění přípony .java a získání relativní cesty
+        relative_path_without_extension = os.path.relpath(file_path, base_directory)[:-5]
+        # Nahrazení oddělovačů cesty tečkami
+        return relative_path_without_extension.replace(os.path.sep, '.')
+    
+    executable_classes = []
+    for java_file in find_java_files(directory):
+        if is_executable_java_file(java_file):
+            executable_classes.append(get_qualified_class_name(java_file, directory))
+    return executable_classes
+
+
+def build_java_project(run_tests: bool) -> BuildResult:
+    output_dir = "output"
+    paths = os.listdir(os.getcwd())
+    pom_xml = [p for p in paths if Path(p).suffix == "pom.xml"]
```

Comment:
```suggestion
    paths = os.listdir(os.getcwd())
```
These two variables were unused.

Reply:
Thank you for you review. I will reflect your notes.

Reply:
Actuallu, the variable paths is alson not used. I will remove it.